src/olympia/accounts/verify.py

```diff
@@ -87,3 +104,48 @@ def get_fxa_profile(token, config):
             )
         )
         raise IdentificationError(f'Could not find profile for {token}')
+
+
+def get_user_token_from_token_data(token_data, config_name):
+    """Return a new (unsaved) FxaToken object from token data."""
+    return FxaToken(
+        access_token=token_data.get('access_token'),
+        access_token_expiry=(
+            datetime.now() + timedelta(seconds=token_data.get('expires_in', 43200))
+        ),
+        refresh_token=token_data.get('refresh_token'),
+        config_name=config_name,
+    )
+
+
+def fxa_access_token_is_valid(user, token_pk):
+    try:
+        token_store = FxaToken.objects.get(user=user, id=token_pk)
+        if token_store.is_expired:
+            config_name = (
+                token_store.config_name
+                if token_store.config_name in settings.ALLOWED_FXA_CONFIGS
+                else settings.DEFAULT_FXA_CONFIG_NAME
+            )
+            # if the access token has expired get a new one
+            token_data = get_fxa_token(
+                refresh_token=token_store.refresh_token,
+                config=settings.FXA_CONFIG[config_name],
+            )
+            new_values = {
+                'access_token': token_data.get('access_token'),
+                'access_token_expiry': datetime.now()
+                + timedelta(seconds=token_data.get('expires_in', 43200)),
+            }
+            if 'refresh_token' in token_data:
+                new_values['refresh_token'] = token_data['refresh_token']
```

Comment:
We currently get a new refresh token don't because we don't pass the required parameter to the endpoint but we _could_.  In theory we could use it to get a new refresh token before it expires (in practise I don't know it there's some shorter validity to a new token).

Reply:
What parameter is that ? The docs aren't super clear what optional params are returned depending on what you pass - I was under the impression that with only a `refresh_token` you can't get a new, fresher `refresh_token`.

Reply:
`access_type=offline` - in theory https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/docs/api.md#post-oauthtoken will return one (I managed to get it to give me a different refresh token during testing, unless I'm misremembering)

Reply:
mmm you shouldn't be able to pass `access_type` to that endpoint with a `refresh_token` according to those docs

Reply:
Trying this just now, I'm getting a 400 error saying `access_type` is an invalid request parameter for that endpoint, as I'd expect

Reply:
🤷 I'll take that line out then